src/adu_workflow/src/agent_workflow.c

```diff
@@ -672,7 +672,10 @@ void ADUC_Workflow_HandleUpdateAction(ADUC_WorkflowData* workflowData)
     if (workflow_isequal_id(workflowData->WorkflowHandle, workflowData->LastCompletedWorkflowId)
         && !workflow_get_force_update(workflowData->WorkflowHandle))
     {
+        char* updateId = workflow_get_expected_update_id_string(workflowData->WorkflowHandle);
         Log_Debug("Ignoring duplicate deployment %s, action %d", workflowData->LastCompletedWorkflowId, desiredAction);
+        ADUC_Workflow_SetInstalledUpdateIdAndGoToIdle(workflowData, updateId);
```

Comment:
This will cause it to report the results approximately every 40 minutes, or whatever the token expiration is. This will cause it to quickly consume message quota--although it would help it to go from in-progress state to a terminal state quicker, it does not address the root issue and has negative side-effect of unnecessarily eating message quota.

The root issue is fixed by PR #659 (just waiting on @nihemstr to address the blocking comment I have), which is to avoid setting to in-progress when processing the root key and to set it to in-progress only once it starts processing an update, which was the previous behavior pre-rootkey package.

If possible, please test out PR #659 to see if addresses the issue in issue https://github.com/Azure/iot-hub-device-update/issues/609

Reply:
ok, thanks. Yes i will try your PR asap.

Reply:
Thanks for trying it out, @JoergZeidler. I will try out the c2d msg from the log.txt to see what the issue was there as soon as I am able to.

Reply:
@jw-msft:
imho, i found the issue in PR #659.

https://github.com/Azure/iot-hub-device-update/blob/50b6ef3a71ca1cdfef70739e1db8a65786ffe1e8/src/adu_workflow/src/agent_workflow.c#L441

should be changed to :
if (IsAducResultCodeFailure(result.ResultCode) || IsAducResultCodeFailure(rootkeyErc.ResultCode))

Reply:
Hi @jw-msft,

I investigate some time to analyse whats going wrong. Imho, you can not check the "rootkeyErc" result here:

https://github.com/Azure/iot-hub-device-update/blob/50b6ef3a71ca1cdfef70739e1db8a65786ffe1e8/src/adu_workflow/src/agent_workflow.c#L441

I changed that line to the old implementation, and at least the normal update process will work again.

Reply:
Thanks for sharing this finding--it is a correct understanding. I have pushed this commit: [34e6d5cd2f25934b9370fc351c9532e663b741ca](https://github.com/Azure/iot-hub-device-update/commit/34e6d5cd2f25934b9370fc351c9532e663b741ca) that also addresses: https://github.com/Azure/iot-hub-device-update/pull/659#discussion_r1785490268